### PR TITLE
[Snippy_Variants and Snippy_Streamline_FASTA] Accept assembly_fasta as input

### LINKED ISSUE #568: [Snippy] Make Snippy_Variants and Snippy_Streamline compatible with FASTA input type
:cool:

### LINKED ISSUE #590: Move the updated user-provided strings out of the input block to prevent user modification
:cool:

### :pushpin: Explain the Request

The updated user-provided strings turn spaces into underscores. This is awesome. However, if someone erroneously provided text in the `_updated` optional input field on Terra, it would render the fun space-removal feature moot. 

We should move all of those updated variables out of the Input block to just beneath them so they are not able to be modified by the user.

### PR DESCRIPTION
**Pull request by cimendes**
This PR closes #568 and closes #590

🗑️ This dev branch should <NOT> be deleted after merging to main.

## :brain: Summary

This PR makes it possible to run **Snippy_Variants** with multiple types of input data:
- Illumina Paired-End ( `"--R1 ~{read1} --R2 ~{read2}"`)
- Illumina Single-End (`"--se ~{read1}"`)
- FASTA (`"--ctgs ~{assembly_fasta}"`)

A new workflow is added, **Snippy_Streamline_FASTA_PHB**, that recreates the Snippy_Streamline workflow but for FASTA input data. 

## :zap: Impacted Workflows/Tasks

The main impacted workflow is **Snippy_Variants_PHB**. All input data types (`read1`, `read2` and `assembly_fasta`) are now optional. The logic to evaluate the provided inputs is within the **Snippy_Variants task**. This is to avoid nesting this task in a conditional which would break the integration in Snippy_Streamline. 

A new workflow was created: **Snippy_Streamline_FASTA_PHB**. There was an attempt to make the original Snippy_Streamline_PHB workflow compatible with multiple input types to mirror Snippy_Variants, but due to constraints in the zip operator, where optional arrays cannot be passed, it required this new behaviour to be implemented in a new workflow. 

This PR may lead to different results in pre-existing outputs: **No**

This PR uses an element that could cause duplicate runs to have different results: **No**

### ➡️ Inputs

- Read1 and Read2 are now optional inputs in Snippy_Variants_PHB
- Assembly_fasta is now a new optional input in Snippy_Variants_PHB

### ⬅️ Outputs

None changed.

## :test_tube: Testing

- **Snippy_Variants with assembly data:** https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/ba65512f-bca6-4391-82b5-5e47ce0a10fe ✅ 
- **Snippy_Streamline_FASTA_PHB:** https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/31f2a961-7bbb-432d-a198-7218a78108bc ✅ 

## :microscope: Final Developer Checklist

- [x] The workflow/task has been tested and results, including file contents, are as anticipated
- [x] The CI/CD has been adjusted and tests are passing (Theiagen developers)
- [x] Code changes follow the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-51b66a47dde54c798f35d673fff80249)
- [x] Documentation and/or workflow diagrams have been updated if applicable (Theiagen developers only)

## 🎯 Reviewer Checklist

- [ ] All changed results have been confirmed
- [ ] You have tested the PR appropriately (see the [testing guide](https://theiagen.notion.site/PR-Testing-Guide-Determining-Appropriate-Levels-of-Testing-4764e98a6aeb460185039c0896714590) for more information)
- [ ] All code adheres to the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-51b66a47dde54c798f35d673fff80249)
- [ ] MD5 sums have been updated
- [ ] The PR author has addressed all comments
- [ ] The documentation has been updated

### CODE REVIEW

**Comment by cimendes** [comment]
@sage-wright documentation has been updated

File: workflows/phylogenetics/wf_snippy_streamline_fasta.wdl [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import "../../tasks/utilities/data_import/task_ncbi_datasets.wdl" as ncbi_datasets_task
+import "../../workflows/phylogenetics/wf_snippy_tree.wdl" as snippy_tree_workflow
+import "../../workflows/standalone_modules/wf_snippy_variants.wdl" as snippy_variants_workflow
+
+workflow snippy_streamline_fasta {
+  input {
+    Array[File] assembly_fasta
+    Array[String] samplenames
+    String tree_name
+    String tree_name_updated = sub(tree_name, " ", "_")
```
Comment (sage-wright) [inline workflows/phylogenetics/wf_snippy_streamline_fasta.wdl:15]: Can you move this out of the input section to the line beneath it so it doesn't show up on Terra?

**Review by sage-wright — CHANGES_REQUESTED** [review]
Looks great, just one minor modification requested to prevent the tree name updated from showing up on Terra -- should have no functional issues.

**Comment by cimendes** [comment]
I've moved the tree_name_updated from the input block on snippy_streamline_fasta and snippy_tree
image

**Review by sage-wright — APPROVED** [review]
wonderful

**Pull request merged**